### PR TITLE
RPi-#10. Controller strategy pattern

### PR DESCRIPTION
**Pull request by FYRichie**
### What is this PR for?

Turn origin controller/main.cpp style into strategy pattern
There are some functions still not implemented in rpiMgr.h/rpiMgr.cpp

### What type of PR is it?

Improvement

### Todos

-   [ rpiMgr.h, rpiMgr.cpp] -   Task

### What is the Github issue?

RPi-#10. Controller strategy pattern

### How should this be tested?

None(only change files in test_controller)

### Questions:

-   Do the license files need updating? No
-   Are there breaking changes for older versions? Yes
-   Does this need new documentation? No

### CODE REVIEW

**Comment by FYRichie** [comment]
@Kenchu123 @ChenWish Can you help me review my code? Thx.

File: setup_zeromq.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+brew install zeromq
```
Comment (Kenchu123) [inline setup_zeromq.sh:1]: `brew install` only works on Mac OSx.
Please change to RPi ways or a more general ways.

File: test_controller/inc/method.h [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+class StatusLight: public BaseMethod {
+public:
+    StatusLight(zmq::socket_t& socket): BaseMethod(socket){};
+    void exec(const vector<string>& cmd, bool& quit){
+        rpiMgr->statuslight();
+    }
+};
+
+// Below methods won't have interaction with interfaces
```
Comment (Kenchu123) [inline test_controller/inc/method.h:85]: I'm not sure what the comments here mean. I think the tests should also be called by the Cli with socket interface?

File: test_controller/src/rpiMgr.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        cerr << "Error: cannot open ./asset/LED.json" << endl;
+    else {
+        infile >> LEDJson;
+        for (json::iterator it = LEDJson.begin(); it != LEDJson.end(); ++it){
+            for (json::iterator it2 = it.value().begin(); it2 != it.value().end(); ++it2){
+                json j;
+                for (int i = 0; i < it2.value().size(); ++i){
+                    uint8_t B = ((unsigned long)it2.value()[i]) % 256;
+                    uint8_t G = (((unsigned long)it2.value()[i]) >> 8) % 256;
+					uint8_t R = (((unsigned long)it2.value()[i]) >> 16) % 256;
```
Comment (Kenchu123) [inline test_controller/src/rpiMgr.cpp:15]: I think there may be some wrong indent. Please fix them.

File: README.md [hunk 4]
```diff
@@ -1 +1,10 @@
-# 2021-LightDance-RPi
\ No newline at end of file
+# 2021-LightDance-RPi
+
+### Setup zeromq
+
+You must first install `cppzmq` to enable the usage of controller
```
Comment (Kenchu123) [inline README.md:5]: Can you have more explanation or links for installing `cppzmq`?

**Pull request merged**